sim/scenarios/corrupt-rate/corrupt-rate-error-model.cc

```diff
@@ -30,6 +30,8 @@ CorruptRateErrorModel::CorruptRateErrorModel()
 void CorruptRateErrorModel::DoReset(void) { }
 
 bool CorruptRateErrorModel::DoCorrupt(Ptr<Packet> p) {
+    if(!IsUDPPacket(p)) return false;
```

Comment:
Add a comment here saying that this applies only to PPP links, since IsUDPPacket assumes it.